docs/docs/building.md

```diff
@@ -4,158 +4,208 @@ title: Building an upgradeable application
-  uint256 public x;
-  
-  function initialize(uint256 _x) isInitializer public {
-    x = _x;
-  }
-  
-  function y() public pure returns (uint256) {
-    return 1337;
-  }
-}
-```
+    export OWNER=<address>
 
-Use truffle to compile the new version of your code:
-```sh
-npx truffle compile
-```
+OK, so we finish this step by linking the standard library and pushing our application to the network:
 
-We'll now use `zos` to register and deploy the new code for `MyContract` to the blockchain. Sync the new version of your project by running: 
+    zos link openzeppelin-zos
+    zos push --network development --deploy-stdlib
 
-```
-zos push --network <network>
-```
+We pass that `--deploy-stdlib` flag because we are using a development network that started clean. When you deploy your application to a real network where the `openzeppelin-zos` standard library has already been deployed, you won't need this flag.
 
-After running this command, the new versions of your project's contracts are deployed in the blockchain. 
-However, the already deployed proxies are still running with the old implementations. You need to upgrade
-each of the proxies individually. To do so, you just need to run this for every contract: 
+The first time you run this command for a specific network, a new
+`package.zos.<network>.json` will be created. This file will reflect the status
+of your project in that network.
 
-```
-zos upgrade <contract_name_1> --network <network>
-zos upgrade <contract_name_2> --network <network>
-...
-zos upgrade <contract_name_n> --network <network>
-```
+TODO: test this and comment about the resulting logs. On the published zos, the set-stdlib command fails, wait for a new release.
 
-In our simple example:
-```
-zos upgrade MyContract --network development
-```
+## Upgrading a contract
 
-Voilà! Your contract has now been upgraded. The address is the same as before, but the code has been changed to the latest version. Repeat the same steps for every code update you want to perform.
+The rules for our basil lights will now be set in stone, enforced by the immutability of the Ethereum blockchain. This sounded great... until we found an embarrassing bug: we are never updating `highestDonation`! The donations would not grow as we expect because the stakes will never be higher than 10 wei. Luckily we caught this before going to mainnet, but are we sure that our contract is bullet proof and something like this will never happen again? The same feature that makes Ethereum secure is now making us feel insecure about our programming abilities and scared of finding a security vulnerability too late.
+
+Fear not, ZeppelinOS allows us to keep the transparency and immutability of a deployed version of a contract, and also to opt-in for a contract in which the owner can upgrade the implementation. This is done through a proxy that forwards the calls to the latest implementation of the contract. To create a proxy for Basil, run:
```

Comment:
", and also to opt-in for ..", I'd change it to: ", but also to opt for.."

Reply:
done